### PR TITLE
Feature/571 custom download widget

### PR DESCRIPTION
**Pull request by cschwinderg**
## Related Issues:
* [HMW-571](https://jira.epa.gov/browse/HMW-571)

## Main Changes:
* Switched to a custom download widget. This custom download widget has all of the same features of the out of the box widget, but with a simplified UI.
  * The print service fails occasionally (the out of the box widget had the same problem), so I added some retry logic. 

## Steps To Test:
1. Run `npm run ci --legacy-peer-deps`
2. Navigate to http://localhost:3000/community/dc/overview
3. Test various features of the download widget
4. Block the `utility.arcgisonline.com/arcgis/rest/services/Utilities/PrintingTools/GPServer/Export%20Web%20Map%20Task/execute` request
5. Verify the retry logic works 

## TODO:
- [ ]  See if Kiki would like to remove any of the features.

### CODE REVIEW

**Review by maxdiebold-erg — CHANGES_REQUESTED** [review]
Nice work! Looks very clean and operates smoothly.

I've made just a couple minor suggestions below.

Additionally, I'm not able to comment on unedited lines, but it looks like calling `render` inside `useMemo` is throwing a warning, so an effect should be used instead in lines 919-944:
```
  // create the download widget
  useEffect(() => {
    if (!view || services.status !== 'success') return;

    const container = document.createElement('div');
    render(<DownloadWidget services={services} view={view} />, container);

    const downloadWidget = new Expand({
      expandIconClass: 'esri-icon-download',
      expandTooltip: 'Open Download Widget',
      collapseTooltip: 'Close Download Widget',
      view,
      mode: 'floating',
      autoCollapse: true,
      content: container,
    });

    view?.ui.add(downloadWidget, { position: 'top-right', index: 3 });

    return function cleanup() {
      if (downloadWidget) view?.ui.remove(downloadWidget);
    };
  }, [services, view]);
```

File: app/client/src/components/shared/MapWidgets.tsx [hunk 1]
```diff
@@ -1922,4 +1930,456 @@ function ShowSelectedUpstreamWatershed({
...
+
+    setScaleWatcherInitialized(true);
+  }, [enableScale, scaleWatcherInitialized, view]);
+
+  useEffect(() => {
+    return function cleanup() {
+      setStatus('idle');
+      setErrorMessage('');
+    };
+  }, []);
```
[the remark this answers is not in the document]
Comment (cschwinderg) [reply app/client/src/components/shared/MapWidgets.tsx:2104]: This was some code that I intended to remove. I'll go ahead and remove that code.

**Comment by cschwinderg** [comment]
@maxdiebold-erg Good catches! I went ahead and implemented your suggestions.

File: app/client/src/components/shared/MapWidgets.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/shared/MapWidgets.tsx]: Oh okay, I actually wasn't worried about that, I was just suggesting that the retry timeout be cleared in that cleanup function.

[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/shared/MapWidgets.tsx]: It doesn't look like clearing the timeout makes a difference either way, though, so I think it's fine as is.

**Comment by maxdiebold-erg** [comment]
Thanks for implementing those. I'll go ahead and review #878 first, since that's included in this PR.

**Comment by cschwinderg** [comment]
@maxdiebold-erg I accidentally pushed up that merge. I meant to just test locally to make sure my accordion changes didn't mess up the accordion in the download widget. 

I reverted those changes, so this PR is only for the download widget.

**Review by maxdiebold-erg — APPROVED** [review]
Ah, okay, in that case everything looks great!

**Pull request merged**